backend/app/services/base.py

```
@@ -0,0 +1,49 @@
+    def get_multi(
+        self, db: Session, *, skip: int = 0, limit: int = 100
+    ) -> List[ModelType]:
+        """Get multiple records with pagination."""
+        return self.repository.get_multi(db, skip=skip, limit=limit)
+
+    def create(self, db: Session, *, obj_in: CreateSchemaType) -> ModelType:
+        """Create a new record."""
+        return self.repository.create(db, obj_in=obj_in)
+
+    def update(
+        self, db: Session, *, id: int, obj_in: UpdateSchemaType | Dict[str, Any]
```

Ruff (0.12.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)